aten/src/ATen/native/cpu/DistanceOpsKernel.cpp

```diff
@@ -130,10 +127,6 @@ struct PDist {
         self_j += m;
         if (self_j == self_end) {
```

Comment:
All this pointer maths smells like a direct port from legacy TH

Reply:
As the original author, I was mostly following what I saw / didn't see other better examples of to handle the arbitrary strides. Not that I'll necessarily have time to rework the batched version, but do you have a better example of an ideal way to handle this?

Reply:
I should flagged this as minor. You could rewrite this to do int64_t comparisons i.e. initialize `j = (l * n + j) * m` and then always do the explicit offset calculation. It'll remove the explicit data dependency, which might help performance wise.

Reply:
I guess I always figured it was "better" to work with pointers to view them like iterators, and that perf would be better with pointers rather than mandating 64bit ints. Note, these thoughts are based on no backing data, other than it generally seems bat to tell the compiler I want 64bit int, when really I want memory addresses and there's no reason to be explicit. Something in the same veing as using size_t instead of in int64.

Reply:
That might be true and probably in particular in the past. It does make it harder to keep track on what's going on though. I'd not use these kind of micro optimizations without careful profiling. Having said that, maybe getting rid of this in this case will make it much worse. In general, I'd expect a modern compiler to know what's the best thing to do here. Something that's harder to figure out however are the exact data dependencies, if you use the same pointer everywhere.